MCP_Server/server.py

```
@@ -498,6 +573,577 @@ def stop_playback(ctx: Context) -> str:
+        ableton = get_ableton_connection()
+        result = ableton.send_command("clear_clip_notes", {
+            "track_index": track_index,
+            "clip_index": clip_index
+        })
+        return f"Cleared {result.get('notes_removed', 0)} notes from clip at track {track_index}, slot {clip_index}"
+    except Exception as e:
+        logger.error(f"Error clearing clip notes: {str(e)}")
+        return f"Error clearing clip notes: {str(e)}"
+
+@mcp.tool()
+def duplicate_clip(ctx: Context, track_index: int, clip_index: int, target_clip_index: int) -> str:
```

Ruff (0.14.14)

[warning] Unused function argument: `ctx`

(ARG001)